pkg/drivers/virtiofs/virtiofs.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2025 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package virtiofs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/google/uuid"
+)
+
+// Mount is a directory on the host shared with the guest using
+// virtiofs.
+type Mount struct {
+	// HostPath is an absolute path to existing directory to share with the guest.
+	HostPath string
+	// GuestPath is a path in the guest for mounting the shared directory.
+	GuestPath string
+	// Tag is a string idenfifying the shared file system in the guest.
+	Tag string
+}
+
+// ValidateMounts parses and validates list of mount strings and returns list of
+// validated mount.
+func ValidateMounts(args []string) ([]*Mount, error) {
+	var mounts []*Mount
+
+	seenHost := map[string]*Mount{}
+	seenGuest := map[string]*Mount{}
+
+	for _, s := range args {
+		mount, err := ParseMount(s)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := mount.Validate(); err != nil {
+			return nil, err
+		}
+
+		if existing, ok := seenHost[mount.HostPath]; ok {
+			return nil, fmt.Errorf("host path %q is already shared at guest path %q", mount.HostPath, existing.GuestPath)
+		}
+		seenHost[mount.HostPath] = mount
+
+		if existing, ok := seenGuest[mount.GuestPath]; ok {
+			return nil, fmt.Errorf("guest path %q is already shared from host path %q", mount.GuestPath, existing.HostPath)
+		}
+		seenGuest[mount.GuestPath] = mount
+
+		mounts = append(mounts, mount)
+	}
+
+	return mounts, nil
+}
+
+// ParseMount parses a string in the format "/host-path:/guest-path"
+// and returns a new Mount instance.
+func ParseMount(s string) (*Mount, error) {
```

Comment:
add unit test for this please, can ask AI to generate unit tests

Reply:
Add tested covering all the code we can use now expect setting up mounts. We can add mount test to the smoke test.